p2p/src/error.rs

```diff
@@ -36,7 +35,6 @@ define_error! {
             | _ | { "public key missing" },
 
         Signature
-            [ DisplayOnly<SignatureError> ]
             | _ | { "signature error" },
```

Comment:
Would it not be worthwhile to bubble up the original error here? Unfortunately it seems like the `ed25519_consensus::Error` type doesn't implement `std::fmt::Display` unless you turn on the `std` feature.

Reply:
The motivation for removing the original error was that the code this is replacing used an opaque error type, so there's no additional information conveyed by the original error, and was easier to just remove it rather than patch the `ed25519_consensus::Error` type to have an extra `Display` impl.  But I could add that `Display` impl in another release.